### PR TITLE
Deprecate some methods in `Client`

### PR DESCRIPTION
**Pull request by danicheg**
Addresses #6650.

### CODE REVIEW

File: client/shared/src/main/scala/org/http4s/client/Client.scala [hunk 1]
```diff
@@ -101,6 +101,7 @@ trait Client[F[_]] {
       d: EntityDecoder[F, A]
   ): F[A]
 
+  @deprecated("Use req.flatMap(expect(_))", "0.23.16")
```
Comment (danicheg) [inline client/shared/src/main/scala/org/http4s/client/Client.scala:104]: If we decide to accept this, in the `1.x` series perhaps we should remove these deprecations then. WDYT?

Comment (armanbilge) [reply client/shared/src/main/scala/org/http4s/client/Client.scala:104]: Yes. In theory it is Nice:tm: to do at least a single deprecation cycle in 1.x as well. But it also means more work for us.

Comment (satorg) [reply client/shared/src/main/scala/org/http4s/client/Client.scala:104]: I wonder how many projects are affected by these deprecations. In a case if there are quite a lot of them then perhaps we could consider preserving such methods in a form of `Client` extension syntax.

Comment (armanbilge) [reply client/shared/src/main/scala/org/http4s/client/Client.scala:104]: I disagree. What is the point of putting it into extension syntax? If we intend to keep it, we should just keep it on `Client`.

Comment (danicheg) [reply client/shared/src/main/scala/org/http4s/client/Client.scala:104]: Agree. I thought that was our intent to clean up API.

**Comment by danicheg** [comment]
So, this got two likes, I think it's legit to move onward. Hope to don't receive a ton of dislikes from users 🤞🏻

File: client/shared/src/main/scala/org/http4s/client/Client.scala (hunk 1 above)
Comment (satorg) [reply client/shared/src/main/scala/org/http4s/client/Client.scala:104]: I would not say I think we should. Personally I would prefer just to get rid of those methods. But wondering if there could be a demand for them from dependent projects.

**Pull request merged**